### PR TITLE
adding tests for Remote Client and GHA Repository

### PR DESCRIPTION
**Pull request by kolioOtSofia**
Closes: #Issue

### CODE REVIEW

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/tearDown.st [hunk 1]
```diff
@@ -0,0 +1,4 @@
+running
+tearDown 
+
+	mockAPI destroy
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/tearDown.st]:
```suggestion
	mockAPI destroy.
	super tearDown
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/testInitialization.st [hunk 2]
```diff
@@ -1,4 +1,8 @@
 tests
 testInitialization
 	
-	self assert: repository client notNil
\ No newline at end of file
+	| newClient |	
+	newClient := ATDDGitHubActionsClient new. 
+	repository client: newClient.
+	self assert: repository client equals: newClient
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/testInitialization.st]:
```suggestion
	| newClient |

	newClient := ATDDGitHubActionsClient new. 
	repository client: newClient.
	self assert: repository client equals: newClient
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/testRequireRepository.st [hunk 3]
```diff
@@ -0,0 +1,5 @@
+tests
+testRequireRepository
+	
+	repository client. 
+	self assert: repository requireRepository equals: (Json readFrom: ATDDGitHubActionMockAPI defaultRuns readStream)
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/testRequireRepository.st]:
```suggestion
tests
testRequireRepository
	
	self assert: repository requireRepository equals: (Json readFrom: ATDDGitHubActionMockAPI defaultRuns readStream)
```
I am not sure if the line I took out does anything or not

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/setUp.st [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	client := ATDDGitHubActionsClient new. 
+	client apiUrl: 'http://localhost:9999/'.
+	gitHubAction 
+		pause;
+		repositoryURL: 'hpi-swa-teaching/AutoTDD'.
+	gitHubAction repository client: client.
+	mockApi := ATDDGitHubActionMockAPI new. 
+	resultsWindow := ATDDGitHubActionResults openForTarget: gitHubAction. 
+	windowsToDelete := OrderedCollection new. 
+	windowsToDelete add: resultsWindow
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/setUp.st]:
```suggestion
running
setUp
	
	| client |
	
	super setUp.
	gitHubAction := ATDDGitHubAction new.
	client := ATDDGitHubActionsClient new.
	client apiUrl: 'http://localhost:9999/'.
	gitHubAction
		pause;
		repositoryURL: 'hpi-swa-teaching/AutoTDD'.
	gitHubAction repository client: client.
	mockApi := ATDDGitHubActionMockAPI new.
	resultsWindow := ATDDGitHubActionResults openForTarget: gitHubAction.
	windowsToDelete := OrderedCollection new.
	windowsToDelete add: resultsWindow
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/tearDown.st [hunk 5]
```diff
@@ -0,0 +1,8 @@
+running
+tearDown
+	
+	super tearDown.
+	gitHubAction delete.
+	mockApi destroy.
+	windowsToDelete do: [ :each | each changed: #close ]
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/tearDown.st]:
```suggestion
running
tearDown

	gitHubAction delete.
	mockApi destroy.
	windowsToDelete do: [:each | each changed: #close ]
	super tearDown.
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testAnsiParsing.st [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	parsedString := resultsWindow parseAnsiCodeOf: sampleString. 
+	self assert: parsedString hasColorAttribute equals: true. 
+	self assert: (parsedString asString includesSubstring: 'ERROR') equals: true.
+	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #error).
+	
+	sampleString := '2021-05-23T12:57:07.9111495Z [0m[33m ✓[0m #testInitialization (0ms)[0m'.
+	parsedString := resultsWindow parseAnsiCodeOf: sampleString. 
+	self assert: parsedString hasColorAttribute equals: true. 
+	self assert: (parsedString asString includesSubstring: 'FAIL') equals: true.
+	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #failure)
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testAnsiParsing.st:22]:
```suggestion
tests
testAnsiParsing

	| sampleString parsedString |

	sampleString := '2021-05-23T12:57:07.9111495Z [0m[32m ✓[0m #testInitialization (0ms)[0m'.
	parsedString := resultsWindow parseAnsiCodeOf: sampleString.
	self assert: parsedString hasColorAttribute equals: true.
	self assert: (parsedString asString includesSubstring: 'PASS') equals: true.
	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #successful).

	sampleString := '2021-05-23T12:57:07.9111495Z [0m[31m ✓[0m #testInitialization (0ms)[0m'.
	parsedString := resultsWindow parseAnsiCodeOf: sampleString.
	self assert: parsedString hasColorAttribute equals: true.
	self assert: (parsedString asString includesSubstring: 'ERROR') equals: true.
	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #error).

	sampleString := '2021-05-23T12:57:07.9111495Z [0m[33m ✓[0m #testInitialization (0ms)[0m'.
	parsedString := resultsWindow parseAnsiCodeOf: sampleString.
	self assert: parsedString hasColorAttribute equals: true.
	self assert: (parsedString asString includesSubstring: 'FAIL') equals: true.
	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #failure)
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testIsSmalltalkCiLog.st [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+	| sampleString |
+	
+	sampleString := 'I dont know, this is some text that is not a valid log'.
+	self assert: (resultsWindow isSmalltalkCILog: sampleString) equals: false.
+	sampleString := 
+	'This is some log, 
+	some more log
+	Run hpi-swa/setup-smalltalkCI
+	some more log'.
+	self assert: (resultsWindow isSmalltalkCILog: sampleString) equals: true
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testIsSmalltalkCiLog.st]:
```suggestion
	sampleString := 'I dont know, this is some text that is not a valid log'.
	self assert: (resultsWindow isSmalltalkCILog: sampleString) equals: false.
	sampleString :=
		'This is some log,
		some more log
		Run hpi-swa/setup-smalltalkCI
		some more log'.
	self assert: (resultsWindow isSmalltalkCILog: sampleString) equals: true
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testParsingSmalltalkCILog.st [hunk 8]
```diff
@@ -0,0 +1,8 @@
+tests
+testParsingSmalltalkCILog
+
+	| parsedResult |
+	parsedResult := resultsWindow parseSmalltalkCILog: self sampleSmalltalkLog.
+	self assert: parsedResult hasColorAttribute equals: true.
+	"does not include time-stamps"
+	self assert: (parsedResult asString includesSubstring: '2021-05-23T12') equals: false.
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testParsingSmalltalkCILog.st]:
```suggestion
tests
testParsingSmalltalkCILog

	| parsedResult |

	parsedResult := resultsWindow parseSmalltalkCILog: self sampleSmalltalkLog.
	self assert: parsedResult hasColorAttribute equals: true.
	"does not include time-stamps"
	self assert: (parsedResult asString includesSubstring: '2021-05-23T12') equals: false.
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/setUp.st [hunk 9]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
+setUp
+
+	client := ATDDRemoteClient new
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/setUp.st]:
```suggestion
running
setUp

	client := ATDDRemoteClient new
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testErrorDictionary.st [hunk 10]
```diff
@@ -0,0 +1,9 @@
+as yet unclassified
+testErrorDictionary 
+
+	| errorDictionary |
+
+	errorDictionary := client errorDictionary.
+	self assert: (errorDictionary at: 401) = ATDDConnectionError.
+	self assert: (errorDictionary at: 403) = ATDDAccessDeniedError. 
+	self assert: (errorDictionary at: 404) = ATDDNotFoundError
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testErrorDictionary.st]:
```suggestion
tests
testErrorDictionary

	| errorDictionary |

	errorDictionary := client errorDictionary.
	self assert: (errorDictionary at: 401) = ATDDConnectionError.
	self assert: (errorDictionary at: 403) = ATDDAccessDeniedError.
	self assert: (errorDictionary at: 404) = ATDDNotFoundError
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testThrowExceptionForResponseCode.st [hunk 11]
```diff
@@ -0,0 +1,7 @@
+as yet unclassified
+testThrowExceptionForResponseCode
+
+	self should: [client throwExceptionForResponseCode: 401] raise: ATDDConnectionError.
+	self should: [client throwExceptionForResponseCode: 403] raise: ATDDAccessDeniedError. 
+	self should: [client throwExceptionForResponseCode: 404] raise: ATDDNotFoundError
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testThrowExceptionForResponseCode.st]:
```suggestion
tests
testThrowExceptionForResponseCode

	self should: [client throwExceptionForResponseCode: 401] raise: ATDDConnectionError.
	self should: [client throwExceptionForResponseCode: 403] raise: ATDDAccessDeniedError. 
	self should: [client throwExceptionForResponseCode: 404] raise: ATDDNotFoundError
```

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/tearDown.st [hunk 12]
```diff
@@ -0,0 +1,5 @@
+running
+tearDown
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/tearDown.st]:
```suggestion
tearDown
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/testInitialization.st [hunk 13]
```diff
@@ -1,4 +1,8 @@
 tests
 testInitialization
 	
-	self assert: repository client notNil
\ No newline at end of file
+	| newClient |
+	
+	newClient := ATDDGitHubActionsClient new.
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsRepositoryTest.class/instance/testInitialization.st]:
```suggestion
	newClient := ATDDGitHubActionsClient new.
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/tearDown.st [hunk 14]
```diff
@@ -0,0 +1,9 @@
+running
+tearDown
+	
+	super tearDown.
+	gitHubAction delete.
+	mockApi destroy.
+	windowsToDelete do: [ :each | each changed: #close ].
+	super tearDown
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/tearDown.st]:
```suggestion
running
tearDown
	
	super tearDown.
	gitHubAction delete.
	mockApi destroy.
	windowsToDelete do: [ :each | each changed: #close ].
	super tearDown
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testIsSmalltalkCiLog.st [hunk 15]
```diff
@@ -0,0 +1,13 @@
+tests
+testIsSmalltalkCiLog
+
+	| sampleString |
+	
+	sampleString := 'I dont know, this is some text that is not a valid log'.
+	self assert: (resultsWindow isSmalltalkCILog: sampleString) equals: false.
+	sampleString :=
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testIsSmalltalkCiLog.st]:
```suggestion
	sampleString :=
```

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testParsingSmalltalkCILog.st [hunk 16]
```diff
@@ -0,0 +1,9 @@
+tests
+testParsingSmalltalkCILog
+
+	| parsedResult |
+	
+	parsedResult := resultsWindow parseSmalltalkCILog: self sampleSmalltalkLog.
+	self assert: parsedResult hasColorAttribute equals: true.
+	"does not include time-stamps"
+	self assert: (parsedResult asString includesSubstring: '2021-05-23T12') equals: false.
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testParsingSmalltalkCILog.st]:
```suggestion
	self assert: (parsedResult asString includesSubstring: '2021-05-23T12') equals: false
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testErrorDictionary.st [hunk 17]
```diff
@@ -0,0 +1,9 @@
+tests
+testErrorDictionary
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testErrorDictionary.st]:
```suggestion
testErrorDictionary
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/setUp.st [hunk 18]
```diff
@@ -0,0 +1,4 @@
+running
+setUp
+
+	client := ATDDRemoteClient new
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/setUp.st]:
```suggestion
running
setUp

	super setUp.
	client := ATDDRemoteClient new
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testThrowExceptionForResponseCode.st [hunk 19]
```diff
@@ -0,0 +1,7 @@
+tests
+testThrowExceptionForResponseCode
+
+	self should: [client throwExceptionForResponseCode: 401] raise: ATDDConnectionError.
+	self should: [client throwExceptionForResponseCode: 403] raise: ATDDAccessDeniedError. 
+	self should: [client throwExceptionForResponseCode: 404] raise: ATDDNotFoundError
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTest.class/instance/testThrowExceptionForResponseCode.st]:
```suggestion
tests
testThrowExceptionForResponseCode

	self should: [client throwExceptionForResponseCode: 401] raise: ATDDConnectionError.
	self should: [client throwExceptionForResponseCode: 403] raise: ATDDAccessDeniedError.
	self should: [client throwExceptionForResponseCode: 404] raise: ATDDNotFoundError
```

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testAnsiParsing.st (hunk 6 above)
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsResultsTests.class/instance/testAnsiParsing.st]:
```suggestion
tests
testAnsiParsing

	| sampleString parsedString |
	
	sampleString := '...'.
	parsedString := resultsWindow parseAnsiCodeOf: sampleString.
	self assert: parsedString hasColorAttribute equals: true.
	self assert: (parsedString asString includesSubstring: 'PASS') equals: true.
	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #successful).
	
	sampleString := '...'.
	parsedString := resultsWindow parseAnsiCodeOf: sampleString.
	self assert: parsedString hasColorAttribute equals: true.
	self assert: (parsedString asString includesSubstring: 'ERROR') equals: true.
	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #error).
	
	sampleString := '...'.
	parsedString := resultsWindow parseAnsiCodeOf: sampleString.
	self assert: parsedString hasColorAttribute equals: true.
	self assert: (parsedString asString includesSubstring: 'FAIL') equals: true.
	self assert: (parsedString colorAt: 3) equals: (resultsWindow colorForStatus: #failure)
```

Replace the dots with the original string though

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

**Review by SebastianWagner2 — APPROVED** [review]

**Review by benjamin-frost — APPROVED** [review]

**Pull request merged**